pkg/test/e2e/default_app_catalog_tests/application_test.go

```diff
@@ -0,0 +1,408 @@
+)
+
+const (
+	projectName = "def-app-catalog-test-project"
+)
+
+func init() {
+	flag.StringVar(&applicationName, "application-name", "", "name of an application from the default app catalog")
+	flag.StringVar(&applicationNamespace, "application-namespace", "", "namespace of an application from the default app catalog")
+	flag.StringVar(&applicationVersion, "application-version", "", "version of an application from the default app catalog")
+	flag.StringVar(&key, "app-label-key", "", "a Kubernetes recommended label used for identifying the name of an application")
+	flag.StringVar(&names, "names", "", "names of the pods of an application from the default app catalog")
+	credentials.AddFlags(flag.CommandLine)
+	jig.AddFlags(flag.CommandLine)
+	logOptions.AddFlags(flag.CommandLine)
+}
+
+func TestClusters(t *testing.T) {
+	if applicationName == "" || applicationVersion == "" {
+		return
+	}
+
+	rawLog := log.NewFromOptions(logOptions)
+	logger := rawLog.Sugar()
+	ctx := context.Background()
+
+	if err := credentials.Parse(); err != nil {
+		t.Fatalf("Failed to get credentials: %v", err)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		t.Fatalf("failed to build config: %v", err)
+	}
+
+	seedClient, err := ctrlruntimeclient.New(config, ctrlruntimeclient.Options{})
+	if err != nil {
+		t.Fatalf("failed to build ctrlruntime client: %v", err)
+	}
+
+	// set the logger used by sigs.k8s.io/controller-runtime
+	ctrlruntimelog.SetLogger(zapr.NewLogger(rawLog.WithOptions(zap.AddCallerSkip(1))))
+
+	client, cleanup, tLogger, err := createUserCluster(ctx, t, logger, seedClient)
+	if cleanup != nil {
+		defer cleanup()
+	}
+
+	if err != nil {
+		t.Fatalf("failed to create user cluster: %v", err)
+	}
+
+	testUserCluster(ctx, t, tLogger, client)
+}
+
+//gocyclo:ignore
+func testUserCluster(ctx context.Context, t *testing.T, log *zap.SugaredLogger, client ctrlruntimeclient.Client) {
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: applicationNamespace,
+		},
+	}
+
+	// Create the namespace in Kubernetes
+	if applicationNamespace != "kube-system" {
+		err := client.Create(ctx, namespace)
+		if err != nil {
+			t.Fatalf("%v", err)
+		} else {
+			log.Infof("Namespace %s created", namespace.Name)
+		}
+	}
+
+	applicationRefName := applicationName
+	if applicationName == "gpu-operator" {
+		applicationRefName = "nvidia-gpu-operator"
+	}
+
+	application := appskubermaticv1.ApplicationInstallation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      applicationName,
+			Namespace: applicationNamespace,
+		},
+		Spec: appskubermaticv1.ApplicationInstallationSpec{
+			Namespace: &appskubermaticv1.AppNamespaceSpec{
+				Name:   applicationNamespace,
+				Create: true,
+			},
+			ApplicationRef: appskubermaticv1.ApplicationRef{
+				Name:    applicationRefName,
+				Version: applicationVersion,
+			},
+		},
+	}
+
+	if applicationName == "cluster-autoscaler" {
```

Comment:
this is not required, it should be part of the defaultValuesBlock of this app

https://github.com/kubermatic/kubermatic/blob/main/pkg/ee/default-application-catalog/applicationdefinitions/cluster-autoscaler.yaml#L40